### PR TITLE
dependencies config: allow dependency periods

### PR DESCRIPTION
**Pull request by decentralion**
This adds support for the dependencies config so that you can specify
how the weight changes over time (e.g. starting minting only at a
particular time) rather than only setting a consistent policy across all
time.

Test plan: Unit tests updated to accomodate the new case.

### CODE REVIEW

File: src/api/dependenciesConfig.js [hunk 1]
```diff
@@ -76,9 +81,8 @@ export type DependencyConfig = {|
   // from the start of the project up until the first specified weightPeriod.
   // If this is set and there are no explicit weightPeriods, this weight will
   // prevail for the whole timeline of the project.
-  // Currently, weight periods aren't yet implemented, so whatever startWeight is
-  // set will apply across all time.
   +startWeight: number,
+  +periods?: $ReadOnlyArray<MintPeriod>,
```
Comment (topocount) [inline src/api/dependenciesConfig.js:85]: What do you think about naming this "additionalPeriods"? It wasn't clear to me that the initial period shouldn't be set here as well

**Review by topocount — APPROVED** [review]
I just have one naming semantics nit. Otherwise it looks good!

File: src/api/dependenciesConfig.js (hunk 1 above)
Comment (decentralion) [reply src/api/dependenciesConfig.js:85]: We could also remove the `startWeight` setting, and just have users set periods if they want to do any minting. It's a simpler API. What do you think?

Comment (hammadj) [reply src/api/dependenciesConfig.js:85]: +1, should simplify to use only periods

File: src/api/dependenciesConfig.js [hunk 2]
```diff
@@ -148,8 +163,13 @@ export function toDependencyPolicy(
...
-  const period = {weight: config.startWeight, startTimeMs: -Infinity};
-  const policy = {address, periods: [period]};
-  return policy;
+  const startPeriod = {weight: config.startWeight, startTimeMs: -Infinity};
+  const processedPeriods = rawPeriods.map(({startTime, weight}) => ({
+    weight,
+    startTimeMs: fromISO(startTime),
+  }));
+  const periods = [startPeriod].concat(processedPeriods);
+  return {address, periods};
```
Comment (hammadj) [inline src/api/dependenciesConfig.js:174]: Removing start period will simplify this greatly as well

File: src/util/timestamp.js [hunk 3]
```diff
@@ -66,3 +68,18 @@ export function validate(timestampMs: number): TimestampMs {
...
+}
+
+export const timestampMsParser: C.Parser<TimestampMs> = C.fmap(
+  C.number,
+  validateTimestampMs
+);
+export const timestampISOParser: C.Parser<TimestampISO> = C.fmap(
+  C.string,
+  validateTimestampISO
+);
```
Comment (hammadj) [inline src/util/timestamp.js:85]: Did something similar in initiatives parser:
```
const TimestampParser = C.fmap(C.string, (t) => toISO(fromISO(t)));
```

Not sure if there's any practical benefit in forcing it to go through our `toISO` function, but thought I'd share.

**Review by hammadj — APPROVED** [review]
Simplifying to only use periods would be cleaner, but looks good otherwise!

**Review by befitsandpiper — APPROVED** [review]

**Comment by decentralion** [comment]
Cool, thanks for the feedback yall. I'm going to merge this as is, and remove the startWeight in a separate PR.

File: src/util/timestamp.js (hunk 3 above)
Comment (decentralion) [reply src/util/timestamp.js]: That's a neat trick, could have made the implementation for https://github.com/sourcecred/sourcecred/pull/2226 a bit more concise. WOuldn't have needed to factor out the validator function.

**Pull request merged**